streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1142,7 +1147,7 @@ void retryWithBackoff(final Map<TaskId, Set<TopicPartition>> tasksToBeCreated) {
                     }
                 }
 
-                if (tasksToBeCreated.isEmpty()) {
+                if (tasksToBeCreated.isEmpty() || backoffTimeMs > rebalanceTimeoutMs) {
```

Comment:
anyways this could be in a separate PR just throwing it out for discussions @sjmittal @dguy @mjsax We can merge the PR as is if other reviewers could give a thumb up.